app/presenters/hyrax/homepage_presenter.rb

```diff
@@ -6,8 +6,19 @@ def initialize(current_ability)
       @current_ability = current_ability
     end
 
+    # @return [Boolean] If the current user is a guest and the always_display_share_button
+    #   is activated, then return true. Otherwise return true if the signed in
+    #   user has permission to create at least one kind of work.
     def display_share_button?
-      Hyrax.config.always_display_share_button? || current_ability.can_create_any_work?
+      (user_unregistered? && Hyrax.config.always_display_share_button?) ||
```

Comment:
Is that feature a bit of a misnomer then? I read "always_display_share_button?" as if this is true then method "display_share_button?" should also be true.